### ISSUE TITLE
Implement Ridge model for direct net-load forecasting

### ISSUE THREAD

**Issue by jesuscontrerasocana** [jesuscontrerasocana/scaffold#9]
## Goal

Implement a Ridge-regression model to forecast `grid_net_kw` directly using the leakage-safe feature helpers from issue #7.

This issue should cover the Ridge model only. Do not add HGBR, probabilistic forecasting, or load/PV decomposition here.

## Model

Use:

```python
sklearn.linear_model.Ridge
```

Target:

```text
grid_net_kw
```

The model should integrate with the existing `Forecaster` interface, train once, persist to disk, reload, and work with the existing simulation/evaluation flow.

## Initial candidate features

Build features explicitly using the generic helpers from issue #7.

### Calendar

Use:

- cyclic time-of-day encoding:
  - `time_of_day_sin`
  - `time_of_day_cos`
- day-of-week as one-hot/dummy variables

Do not include month initially.

Calendar encoding is model-specific and should remain in the Ridge implementation rather than the generic feature helpers.

### Historical net load

Candidate features:

- lag 15 minutes
- lag 1 hour
- lag 1 day
- lag 1 week
- rolling mean over previous 1 hour
- rolling mean over previous 4 hours

### Known future information

- `most_recent_load_factor_forecast`

This should be appended explicitly from the future exogenous data available to the forecaster.

## Feature selection

Use lightweight built-in scikit-learn feature selection rather than implementing custom selection logic.

Use:

```python
sklearn.feature_selection.SequentialFeatureSelector
```

with Ridge as the estimator.

Feature selection must be time-series safe. Use:

```python
sklearn.model_selection.TimeSeriesSplit
```

rather than ordinary shuffled cross-validation.

For example:

```python
SequentialFeatureSelector(
    Ridge(alpha=1.0),
    direction="forward",
    scoring="neg_mean_absolute_error",
    cv=TimeSeriesSplit(...),
)
```

Keep the implementation simple:

- always retain the calendar representation;
- run feature selection over the main continuous candidate features:
  - lags
  - rolling means
  - `most_recent_load_factor_forecast`
- avoid selecting individual day-of-week dummy columns independently;
- do not add exhaustive subset search, RFE, RFECV, Lasso selection, or a feature-selection framework.

Persist or otherwise record the selected feature set so inference after reload uses exactly the same inputs.

## Feature standardization

Standardize Ridge inputs so that features with different numerical scales are regularized comparably.

Use scikit-learn's:

```python
StandardScaler()
```

Fit the scaler on training data only and reuse the fitted scaler unchanged at inference.

A simple sklearn pipeline is preferred, for example:

```python
Pipeline([
    ("scaler", StandardScaler()),
    ("ridge", Ridge(alpha=1.0)),
])
```

For simplicity, scaling all final Ridge input columns is acceptable, including dummy variables. Do not add unnecessary preprocessing abstractions solely to treat dummy columns differently.

The same scaling treatment must be used during feature selection and final model fitting so that Ridge coefficients and regularization are comparable across candidate features.

## Training

Implement a small model class, for example:

```python
class RidgeNetLoadModel:
    ...
```

Training should:

- use only the training period;
- construct leakage-safe historical features;
- use the same feature definitions at training and inference;
- remove rows where required historical features are unavailable;
- one-hot encode day of week;
- standardize model inputs with a training-fitted `StandardScaler`;
- perform sequential feature selection using `TimeSeriesSplit` with the same scaling treatment;
- fit the final Ridge model using the selected feature set.

Use a simple fixed Ridge regularization parameter initially, e.g.:

```python
Ridge(alpha=1.0)
```

Do not add hyperparameter optimization in this issue.

## Leakage requirements

This is especially important because the forecast horizon can extend to 132 quarter-hours / 33 hours.

At a decision time `t`:

- realized `grid_net_kw` at `t` or later must never be used;
- lag or rolling features that require observations after `t` must remain unavailable;
- only exogenous information explicitly available at prediction time may be used;
- training examples must reproduce the information boundary that exists at inference.

Do not construct training rows from the full dataset in a way that gives long-horizon targets access to realized intermediate net load.

## Inference

The model should:

- accept the same history/future context as existing forecasters;
- construct the appropriate Ridge features;
- apply the persisted scaler/preprocessing and selected feature set;
- return:

```text
net_kw
```

for the requested horizon.

## Persistence

Persist enough state to reproduce predictions exactly after reload:

- fitted Ridge model;
- fitted scaler/preprocessor;
- one-hot encoding state;
- selected feature names;
- any relevant model metadata.

A simple structure is sufficient:

```text
models/ridge/
    forecaster.json
    model.joblib
```

Do not preserve compatibility with legacy Ridge artifacts.

## Model inspection

Make the fitted Ridge model easy to inspect after training.

Expose or save a compact summary containing at least:

- selected feature names;
- fitted coefficients mapped to feature names;
- intercept;
- Ridge `alpha`;
- training MAE and RMSE;
- validation / feature-selection MAE and RMSE where available;
- number of training samples used;
- number of rows dropped because required features were unavailable.

Because inputs are standardized, clearly state that the reported coefficients are in standardized-feature space. This also makes coefficient magnitudes more directly comparable as an inspection aid.

Prefer a simple table / DataFrame or small `summary()` method over custom reporting or plotting.

For example:

```text
feature                              coefficient
lag_1week                             8.41
lag_1day                              4.92
most_recent_load_factor_forecast     -3.76
rolling_mean_4h                       2.18
```

A method such as:

```python
model.summary()
```

returning a dict or DataFrame is sufficient.

## Forecaster integration

Add `ridge` as a selectable model alongside the existing models:

```text
scaffold
weekly
ridge
```

Keep the harness-facing `Forecaster` API unchanged.

## Evaluation

The Ridge model must work with the existing lightweight forecast evaluator so it can be compared directly with the scaffold and weekly baselines.

No new evaluation framework is needed.

Useful comparison metrics remain:

- MAE
- RMSE
- bias
- nMAE
- lead-time error breakdown

## Tests

Add focused tests covering:

- Ridge training and prediction;
- correct output index and `net_kw` column;
- save/load preserves predictions;
- selected features are persisted and reused;
- fitted scaling state is persisted and reused;
- scaling is fitted only on training data;
- no future realized net load is used;
- long-horizon inference remains leakage-safe;
- day-of-week encoding is stable between training and inference;
- model inspection exposes selected features, standardized coefficients, and basic train/validation errors;
- the model works with the existing forecast evaluator.

## Constraints

- Keep the implementation small and explicit.
- Reuse the feature helpers from issue #7.
- Use standard scikit-learn components where possible.
- No abstract model hierarchy.
- No generic feature-selection framework.
- No hyperparameter search.
- No HGBR.
- No probabilistic model.
- No load/PV decomposition.
- Preserve the behavior of the scaffold and weekly models.

## Acceptance criteria

- `ridge` can be selected through the existing training flow.
- Ridge trains using the agreed candidate features.
- Ridge inputs are standardized using a scaler fitted only on training data.
- `SequentialFeatureSelector` with `TimeSeriesSplit` selects among the continuous candidate features using the same scaling treatment.
- Calendar features remain available without arbitrary dummy-column elimination.
- The selected feature set and fitted scaler are persisted.
- A reloaded Ridge model reproduces predictions.
- Forecasts are leakage-safe across the full 33-hour horizon.
- The fitted model’s selected features, standardized coefficients, and basic train/validation errors can be inspected easily.
- The model can be evaluated directly against the existing baselines.

**Comment by jesuscontrerasocana** [comment]
Methodological note: make the multi-step forecasting strategy explicit. With a horizon up to 132 quarter-hours, short lags and rolling features are not directly observed for later forecast steps. Use recursive prediction: predict the next quarter, append that prediction to a working history, rebuild features, and continue step by step. Never substitute future realized `grid_net_kw` for those intermediate values.

**Comment by jesuscontrerasocana** [comment]
Also keep standardization inside the cross-validation folds used by `SequentialFeatureSelector`. The estimator passed to SFS should therefore be a pipeline such as `Pipeline([("scaler", StandardScaler()), ("ridge", Ridge(alpha=1.0))])`, so the scaler is fitted separately on each training fold and does not see validation-fold data.
